src/python/dxpy/bindings/apollo/vizserver_filters_from_json_parser.py

```diff
@@ -522,3 +529,33 @@ def build_partial_overlap_genobin_filters(
         # }
 
         return filter_structure
+
+    def merge_duplicate_filters(self, compound_block):
+        """
+        Recursively walk the compound tree and merge duplicate 'filters' blocks
+        at the same compound level.
+        """
+        if "compound" in compound_block:
+            new_compound = []
+            merged_filters = {}
+
+            for item in compound_block["compound"]:
+                self.merge_duplicate_filters(item)
+
+                if "filters" in item and len(item) == 1:
+                    # Merge filters into the accumulating dict
+                    for key, conditions in item["filters"].items():
+                        merged_filters[key] = []
+                        merged_filters[key].extend(conditions)
+                else:
+                    if merged_filters:
+                        new_compound.append({"filters": merged_filters})
+                        merged_filters = {}
+                    new_compound.append(item)
+
+            if merged_filters:
+                new_compound.append({"filters": merged_filters})
+
+            compound_block["compound"] = new_compound
+
+        return compound_block
```

Comment:
May I please ask you to add a small unit test for this? Also, could you add a concise example to the function description, to make it clearer what gets transformed into what? Could be helpful in the future.